pkg/kubectl/cmd/edit.go

```diff
@@ -228,6 +229,12 @@ func RunEdit(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []strin
 			if err := kubectl.CreateOrUpdateAnnotation(cmdutil.GetFlagBool(cmd, cmdutil.ApplyAnnotationsFlag), updates); err != nil {
 				return preservedFile(err, file, out)
 			}
+			if cmdutil.GetRecordFlag(cmd) {
```

Comment:
Like apply, we should add/update the annotation if either the flag is set or the annotation already exists.